net/FlatBuffers/Table.cs

```diff
@@ -83,15 +84,15 @@ protected int __vector(int offset)
         }
 
         // Initialize any Table-derived type to point to the union at the given offset.
-        protected TTable __union<TTable>(TTable t, int offset) where TTable : Table
+        public T __union<T>(int offset) where T : struct, Accessor
```

Comment:
So nothing is ever cast to `Accessor` or transferred to a variable of that type, so maybe this is safe. It invokes a method coming from Accessor, but still on its own type (inside `__union`). It is then later cast to a `Nullable` of its own type (when `__union` returns).

Yes, running your benchmark would be helpful. One thing that worries me that even if the benchmark says speed is good, do we know this is for all implementations (MS vs mono)? And maybe my code is faster than the old code because of struct use, but still masks an allocation for unions.